src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -103,6 +103,16 @@ void publish_commit_completion(rados_completion_t completion, void *arg) {
     }
 };
 
+static inline std::string get_event_detail_str(
+    rgw::notify::event_entry_t& event_entry) {
+  return fmt::format(
+      " for endpoint: '{}' for bucket_owner: '{}' for bucket: '{}' for object: "
+      "'{}' for event: '{}' ",
+      event_entry.push_endpoint, event_entry.event.bucket_ownerIdentity,
+      event_entry.event.bucket_name, event_entry.event.object_key,
+      event_entry.event.eventName);
+}
```

Comment:
nit: since this was made for dout, consider using a stream output operator to avoid allocating the temporary std::string

```c++
std::ostream& operator<<(std::ostream& out, const event_entry_t& e) {
  return out << "endpoint: '" << e.push_endpoint
      << "', bucket_owner: '" << e.event.bucket_ownerIdentity
      << "', bucket: '" << e.event.bucket_name
      << "', object: '" << e.event.object_key
      << "', event: '" << e.event.eventName << "'";
}
```

Reply:
I think that's discretely more costly than fmt?

Reply:
all of these arguments appear to be strings, so `operator<<` is just copying each of them directly into the streambuf

while fmt may be better optimized for formatting things like ints/floats, i don't see what advantage it offers here that would outweigh the cost of memory allocation

if we wanted the end result in a std::string that we'd have to allocate anyway, then the fmt version is surely better than formatting with a stringstream

but because we're streaming this result like `<< get_event_detail_str(event_entry) <<`, the use of fmt here seems like an anti-pattern

Reply:
there might be a way to get `fmt::format_to()` to copy directly to our streambuf too? that could be the best of both worlds

Reply:
@cbodley, overloaded the `operator<<` for `struct event_entry_t`